packages/metro-config/src/defaults/__tests__/exclusionList-test.js

```diff
@@ -25,14 +25,84 @@ describe('exclusionList', () => {
     path.sep = originalSeparator;
   });
 
-  it('converts forward slashes in the RegExp to the OS specific path separator', () => {
+  it('proves we can write to path.sep for setting up the tests', () => {
     // $FlowFixMe: property sep is not writable.
     path.sep = '/';
+    expect(require('path').sep).toBe('/');
+    // $FlowFixMe: property sep is not writable.
+    path.sep = '\\';
+    expect(require('path').sep).toBe('\\');
+  });
+
+  it('converts forward slashes in the RegExp to the OS specific path separator in macOS/linux', () => {
+    // $FlowFixMe: property sep is not writable.
+    path.sep = '/';
+    // Simple case
     expect('a/b/c').toMatch(exclusionList([new RegExp('a/b/c')]));
+    expect('a/b/c').toMatch(exclusionList([/a\/b\/c/]));
+    // Regular expression that already considered OS specific path separator.
+    // When explictly construct RegExp instance, the string needs to be escaped first.
+    // eg. /.*[/\\]foo[/\\]bar/ => '.*[/\\\\]foo[/\\\\]bar'
+    expect('/foo/bar').toMatch(exclusionList([new RegExp('.*[/\\\\]foo[/\\\\]bar')]));
+    expect('/foo/bar').toMatch(exclusionList([/.*[/\\]foo[/\\]bar/]));
+  });
 
+  it('converts forward slashes in the RegExp to the OS specific path separator in windows', () => {
     // $FlowFixMe: property sep is not writable.
     path.sep = '\\';
-    expect(require('path').sep).toBe('\\');
+    // Simple case
     expect('a\\b\\c').toMatch(exclusionList([new RegExp('a/b/c')]));
+    expect('a\\b\\c').toMatch(exclusionList([/a\/b\/c/]));
+    // Regular expression that already considered OS specific path separator.
+    // When explictly construct RegExp instance, the string needs to be escaped first.
+    // eg. /.*[/\\]foo[/\\]bar/ => '.*[/\\\\]foo[/\\\\]bar'
+    expect('\\foo\\bar').toMatch(exclusionList([new RegExp('.*[/\\\\]foo[/\\\\]bar')]));
+    expect('\\foo\\bar').toMatch(exclusionList([/.*[/\\]foo[/\\]bar/]));
+  });
+
+  it('converts forward slashes in the string to the OS specific path separator in macOS/linux', () => {
+    // $FlowFixMe: property sep is not writable.
+    path.sep = '/';
+    // Simple case
+    expect('a/b/c').toMatch(exclusionList(['a/b/c']));
+    // Strings that already considered OS specific path separator.
+    expect('/foo/bar').toMatch(exclusionList(['.*[/\\]foo[/\\]bar']));
+  });
```

Comment:
This already seems to be tested in `converts forward slashes in the RegExp to the OS specific path separator in macOS/linux`. I personally prefer the split but we should then remove it from the other test case.

Reply:
This test is for string input.